doc/manual/expressions/debug-attributes.xml

```diff
@@ -0,0 +1,47 @@
+<section xmlns="http://docbook.org/ns/docbook"
+      xmlns:xlink="http://www.w3.org/1999/xlink"
+      xmlns:xi="http://www.w3.org/2001/XInclude"
+      version="5.0"
+      xml:id="ssec-debug-attrsets">
+
+<title>Debugging Attribute Sets</title>
+
+<para>Attribute sets are the core of nix, and sometimes can get quite complex.
+During development it may be useful to use
+<function>builtins.toJSON</function>, <function>builtins.toJSON</function> and
+<function>builtins.trace</function> to inspect attribute sets.
+
+<screen>
+$ cat dbg-ex.nix 
+rec {
+  set = { 
+    a = 1;
+    y = "foo";
+    x.z = "0";
+  }; 
+
+  # inspect set
+  json = builtins.toJSON set; 
+  xml = builtins.toXML set; 
+
+  # trace when set is being assigned
+  superSet = { 
+    set = builtins.trace "assigning set!" set;
+    poit = "narf";
+  };
+}
+
+$ nix-instantiate --eval -A json dbg-ex.nix 
+"{\"a\":1,\"x\":{\"z\":\"0\"},\"y\":\"foo\"}"
```

Comment:
hm, I think when we talk about debugging, nix-instantiate --json --eval -A set dbg-ex.nix gives better output.
Maybe even mention python-prettyfying (nix-instantiate --json --eval -A set dbg-ex.nix | python -m 'json.tool' gives a nice output, I learnt that one on the sprint)